### PR TITLE
Added undo/redo to script editor

### PR DESCRIPTION
**Pull request by Leo40Git**
Fixes issue #7.

### CODE REVIEW

File: src/ca/noxid/lab/script/TscPane.java [hunk 1]
```diff
@@ -293,6 +297,52 @@ public void caretUpdate(CaretEvent eve) {
 				}
 			}
 		});
+
+		undoManager = new UndoManager();
+		getStyledDocument().addUndoableEditListener(new UndoableEditListener() {
+			@Override
+			public void undoableEditHappened(UndoableEditEvent e) {
+			if (!e.getEdit().getPresentationName().equals("style change") && recordUndoes) {
+				System.out.println("Add edit: " + e.getEdit().getPresentationName());
```
Comment (taedixon) [inline src/ca/noxid/lab/script/TscPane.java]: you should probably remove this debug statement, I know I have like more than a few lying around but this one in particular will put a lot of lines on the output.

**Comment by taedixon** [comment]
This is alright, but do you think it's best to make each character typed its own undo/redo? It's the most straightforward solution but it means you have to press the button a lot. Is there any way to set it so that it only records significant edits at word/command boundaries? See https://github.com/taedixon/boosters-lab/blob/master/src/ca/noxid/lab/mapdata/MapInfo.java#L785 for an example of how I use non-significant edits to undo a batch of small entityEdit movements when shifting NPCs on a map.

**Comment by Leo40Git** [comment]
Sorry noxid, but I literally have no idea how to do that. Although on my side it's already doing the "each char is it's own undo/redo" thing.

File: src/ca/noxid/lab/script/TscPane.java (hunk 1 above)
Comment (Leo40Git) [inline src/ca/noxid/lab/script/TscPane.java]: Ok sure

**Comment by taedixon** [comment]
Well the implication here was that "maybe that isn't the best approach actually"
http://docs.oracle.com/javase/7/docs/api/javax/swing/undo/UndoableEdit.html#addEdit%28javax.swing.undo.UndoableEdit%29
it seems you can merge edits together. Try this as a starting point.

**Comment by Leo40Git** [comment]
I'll try, probably won't succeed tho

**Comment by Leo40Git** [comment]
okay, i most likely will never complete this, and someone else can probably do it better than me.

**Pull request closed without merging**